### ISSUE TITLE
Cannot extract tarballs containing paths greater than 100 characters

### ISSUE THREAD

**Issue by deciduously** [tangramdotdev/tangram#404]
The `async-tar` crate cannot handle paths greater than 100 characters. @endbr64 noted this similar bug: https://github.com/dignifiedquire/async-tar/issues/39

Here is an example `tree` of an input directory:
```
❯ tree fake-gcc/
fake-gcc
└── even
    └── more
        └── nesting
            └── to
                └── make
                    └── this
                        └── path
                            └── way
                                └── too
                                    └── long
                                        └── libstdc++-v3
                                            └── testsuite
                                                └── 26_numerics
                                                    └── random
                                                        └── chi_squared_distribution
                                                            └── requirements
                                                                └── file.c
```
                                                                
and the resulting checkout of extracting a tarball of it:
```
❯ tree checkout/
checkout
└── fake-gcc
    └── even
        └── more
            └── nesting
                └── to
                    └── make
                        └── this
                            └── path
                                └── way
                                    └── too
                                        └── long
                                            └── libstdc++-v3
                                                └── testsuite
                                                    └── 26_numerics
                                                        └── random
```
The file was created, but at a truncated path.

**Comment by endbr64** [comment]
I put up a WIP fix in #406 that should unblock you. I plan to clean it up and push it upstream tomorrow.

**Comment by deciduously** [comment]
Confirmed - GCC unpacked and compiled using this branch. Thanks!
